pytorch_toolkit/ote/ote/modules/config_transformers/nncf_config_transformer.py

```diff
@@ -0,0 +1,115 @@
+"""
+ Copyright (c) 2020 Intel Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+from copy import copy
+from mmcv import Config
+
+from ote.utils import load_config
+from ..registry import CONFIG_TRANSFORMERS
+from .base import BaseConfigTransformer
+from .utils import merge_dicts_and_lists_b_into_a
+
+def _convert_value_to_bool(v, key):
+    if isinstance(v, bool):
+        return v
+    if isinstance(v, str):
+        if v.lower().strip() in ("on", "true", "yes", "y"):
+            return True
+        if v.lower().strip() in ("off", "false", "no", "n"):
+            return False
+    elif isinstance(v, int):
+        if v == 1:
+            return True
+        if v == 0:
+            return False
+    err_str = f'Cannot convert to bool the value `{v}`'
+    if key is not None:
+        err_str += ' for the key "{key}"'
+    raise RuntimeError(err_str)
+
+@CONFIG_TRANSFORMERS.register_module()
+class NNCFConfigTransformer(BaseConfigTransformer):
+    POSSIBLE_NNCF_PARTS = ('int8', 'sparsity', 'pruning')
+    COMPRESSION_CONFIG_KEY = 'compression_config'
+
+    def __init__(self):
+        super().__init__()
+
+    def process(self, template_path, config_path):
```

Comment:
I didn't find any test that verifies the passed `config.py`, converts-merges it, and compares it with the expected NNCF JSON config.

Reply:
It may be a good idea to implement such a test, but I'm not sure that it is required.

@Ilya-Krylov, what is your opinion?

Reply:
It's reasonable to have such test, I guess @AlexKoff88 requires it, I don't mind

Reply:
I added it to my TODO-s.

I'll implement it in the next PR.